### PR TITLE
Improvements to documentation and docker setups from my getting spun up

### PR DESCRIPTION
**Pull request by tfoote**
This is an aggregate of my making progress on learning how things are working and trying to streamline things a bit. 

The usage of rocker requires a new feature I just added to work around the upstream image setup: https://github.com/osrf/rocker/pull/164

### CODE REVIEW

File: docker/tests/Dockerfile
Comment (chapulina) [inline docker/tests/Dockerfile:77]: I don't see the test output on GitHub actions, did it run and the output got swallowed?

**Review by chapulina — COMMENTED** [review]
> The usage of rocker requires a new feature I just added 

Do we need to wait for a release before merging this PR?

File: docker/tests/Dockerfile
Comment (tfoote) [reply docker/tests/Dockerfile:77]: Ahh, the actions aren't actually running the image, only building the image. I'll add it to the action.

File: docker/build_and_run_docker.sh
Comment (tfoote) [reply docker/build_and_run_docker.sh:32]: Good point. I've homogenized the rocker and docker content.

**Comment by tfoote** [comment]
I have released rocker 0.2.7 to get the new command line argument

File: README.md
Comment (chapulina) [inline README.md:28]: Nitpick: this project doesn't use ROS, so users landing here may not have ROS packages setup. They may setup it at later stages to get `colcon`, but up here on the README, it may not be clear how to set it up.

I always prefer to err on the side of explicitness in this situation to avoid confusing new users who may not be familiar with ROS packages. So I'd suggest either removing the part about "_debian package `python3-rocker` with the ROS packages_" and letting the `rocker` entry page explain how to set it up, or repeating the step-by-step here on the README.

File: README.md
Comment (chapulina) [inline README.md:52]: The launch file was removed in #102 , we should use plain `ign gazebo` commands now

File: docker/tests/Dockerfile
Comment (chapulina) [inline docker/tests/Dockerfile:45]: Is it worth it cleaning it here, considering we'll do an update below again?

**Review by chapulina — APPROVED** [review]
LGTM, I just have minor comments and I think we should wait for @arjo129 's approval too.

File: docker/debug_integration.sh
Comment (arjo129) [reply docker/debug_integration.sh:36]: Good News: The system launches correctly with the correct panes.

Bad News: I may be doing something wrong but when I run via rocker I get the following error:
```
[GUI] [Err] [Ogre2RenderEngine.cc:905]  Unable to create the rendering window: OGRE EXCEPTION(3:RenderingAPIException): OpenGL 3.3 is not supported. Please update your graphics card drivers. in GL3PlusRenderSyste
m::initialiseContext at /var/lib/jenkins/workspace/ogre-2.2-debbuilder/repo/RenderSystems/GL3Plus/src/OgreGL3PlusRenderSystem.cpp (line 3364)                                                                       
[GUI] [Err] [Ogre2RenderEngine.cc:913] Unable to create the rendering window after [11] attempts.                                                                                                                   
[GUI] [Err] [Ogre2RenderEngine.cc:834] Failed to create dummy render window.
```

Might be some (nvidia?) driver issue or a problem with the way I setup rocker :thinking: . 
Also is the `--nvidia` really nessecary? What about scenarios where we aren't using an nvidia card. I don't think we really need cuda acceleration.

**Review by arjo129 — CHANGES_REQUESTED** [review]
Thanks for these changes.

I've been unable to get rocker working on my comp correctly. Could be some GPU setup issue. I'm worrying if we really need the `--nvidia` flag. There are some minor knits in regards to style.

**Comment by tfoote** [comment]
>Is it worth it cleaning it here, considering we'll do an update below again?

Yes, the standard practice when building dockerfiles is to clear the cache at every layer. This has two major reasons. First is that it keeps the image size down. Every file change in the system at the end of a step is saved into the layer. So if you do 3 apt-get updates in three consecutive layers you will get three copies of the cache in each layer assuming they're slightly different because they were pulled at different times. And secondly, because docker caches previous layers, you have no knowledge of how long ago the last update was so you are going to have to do another update. And the cost of the incremental update versus a fresh update is small, and gives you a much more consistent result.

**Comment by mabelzhang** [comment]
Butting in with a nitpick request in my head for a while - can we rename this image so that it doesn't start with `mbari`? Technically, this is not mbari's image, so it might be called `osrf_lrauv` or something other than MBARI LRAUV.

Confusion arises because we host a different, official image on MBARI GitHub that has the MBARI code in it: https://hub.docker.com/r/mbari/lrauv-ignition-sim When you pull it, it's called `mbari/lrauv-ignition-sim`. At the end of the day, I end up with a handful of images under my `docker images`, all called some combination of MBARI LRAUV (the official no-MBARI-source one which I push to DockerHub, the with-source one that's a dependency, and my personal development one. Not counting this one in the osrf repo).

The more we can disambiguate these names, the less confusing it will be for ourselves.

**Comment by tfoote** [comment]
Good idea @mabelzhang I've updated it with the osrf_lrauv to mirror the repo name for clarity.

**Comment by tfoote** [comment]
I worked with @arjo129 on slack and we isolated his crashing to be related to the integrated AMD card getting selected and his `prime-select` settings seemingly not going into the environment.

**Review by chapulina — CHANGES_REQUESTED** [review]

**Comment by tfoote** [comment]
I've rebased this for garden. 

With the new rebuilds of the garden image with different UIDs it needs https://github.com/osrf/rocker/pull/168

File: docker/tests/Dockerfile
Comment (tfoote) [reply docker/tests/Dockerfile:46]: Yes, this removes the contents of the apt cache from the layer, making the docker container much slimmer. Otherwise you just end up with outdated caches in every layer of the container.

**Comment by hidmic** [comment]
> I worked with @arjo129 on slack and we isolated his crashing to be related to the integrated AMD card getting selected and his `prime-select` settings seemingly not going into the environment.

Circling back here. I did not find anything particularly unusual in `rocker`. @arjo129 I'll assume that you've `prime-select`'ed `nvidia`. Could it be an issue with your `LD_LIBRARY_PATH` (https://github.com/NVIDIA/nvidia-docker/issues/612)? Would you mind giving it a shot?

**Comment by caguero** [comment]
Closing this PR for now.

**Pull request closed without merging**
